owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt

```diff
@@ -429,8 +466,15 @@ class LoginActivity : AppCompatActivity(), SslUntrustedCertDialog.OnSslUntrusted
      */
     private fun exchangeAuthorizationCodeForTokens(authorizationCode: String) {
         server_status_text.text = getString(R.string.auth_getting_authorization)
-        val clientAuth =
+
+        val clientRegistrationInfo = oauthViewModel.registerClient.value?.peekContent()?.getStoredData()
+
+        val clientAuth = if (clientRegistrationInfo?.clientId != null && clientRegistrationInfo.clientSecret != null) {
+            OAuthUtils.getClientAuth(clientRegistrationInfo.clientSecret as String, clientRegistrationInfo.clientId)
```

Comment:
About this `as String`, please have a look at https://kotlinlang.org/docs/reference/typecasts.html#unsafe-cast-operator, i think that it could throw an exception

![Screenshot 2021-02-10 at 12 14 17](https://user-images.githubusercontent.com/15261363/107502859-85916a00-6b99-11eb-84c1-d27b448ae825.png)

Reply:
We are checking that it is not null in the line above. So it is not null, thanks for it anyway 👍